sr_hand_health_report/src/sr_hand_health_report/overrun_check.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import rospy
+from sr_hand_health_report.sr_hand_health_report_check import SrHealthReportCheck
+from diagnostic_msgs.msg import DiagnosticArray
+from sr_robot_msgs.msg import EthercatDebug
+
+
+class OverrunCheck(SrHealthReportCheck):
+
+    def __init__(self, hand_side, fingers_to_test):
+        super().__init__(hand_side, fingers_to_test)
+
+        self.check_time = 10
+        self.number_of_drops = 0
+        self.iterations = 0
+        self.overrun_average = 0
+        self.drop_average = 0
+        self._result = None
+
+        rospy.Subscriber("/diagnostics_agg", DiagnosticArray, self.overruns_callback)
+        rospy.Subscriber(f"/{self._hand_prefix}/debug_etherCAT_data", EthercatDebug, self.drops_callback)
+
+    def overruns_callback(self, data):
+        overrun = OverrunCheck.get_recent_overruns_by_regex(data)
+
+        self.overrun_average += int(float(overrun))
+        self.drop_average += self.number_of_drops
+        self.number_of_drops = 0
+        self.iterations += 1
+
+    def drops_callback(self, data):
+        if data.sensors[10] == 0:
+            self.number_of_drops += 1
+
+    @staticmethod
+    def get_recent_overruns_by_regex(msg):
```

Comment:
Where is the regex in this method?

Reply:
I copied this from https://github.com/shadow-robot/common_resources/blob/noetic-devel/sr_utilities_common/scripts/sr_utilities_common/overrun_experiments.py#L26 but still a valid point. Renamed the method.